python_modules/dagster/dagster/grpc/client.py

```diff
@@ -303,6 +304,20 @@ def external_sensor_execution(self, sensor_execution_args, timeout=DEFAULT_GRPC_
 
         return "".join([chunk.serialized_chunk for chunk in chunks])
 
+    def external_notebook_data(self, notebook_data_args, timeout=DEFAULT_GRPC_TIMEOUT):
+        check.inst_param(notebook_data_args, "notebook_data_args", NotebookDataArgs)
+
+        res = self._streaming_query(
+            "ExternalNotebookData",
+            api_pb2.ExternalNotebookDataRequest,
+            timeout=timeout,
+            serialized_external_notebook_data_args=serialize_dagster_namedtuple(notebook_data_args),
+        )
+
+        chunks = list(res)
+
+        return "".join([chunk.serialized_chunk for chunk in chunks])
```

Comment:
I'm not sure we want to do this hand rolled streaming jank anymore - context in https://dagster.phacility.com/D9094

Reply:
streaming might make sense if we can stream the bytes directly to the http response - but if we have to buffer it all then it seems dubious